File: suc/kotlin/083.kt
package com.example.myapplication
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Menu
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)
        windowInsetsController.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())

        setContent {
            MaterialTheme(
                colorScheme = androidx.compose.material3.lightColorScheme(
                    background = Color.White,
                    surface = Color.White,
                    primary = Color(0xFF2196F3), // Blue
                    onPrimary = Color.White,
                    secondary = Color(0xFFECEFF1),
                    onSecondary = Color.Black
                )
            ) {
                ServerStatusScreen()
            }
        }
    }
}
@Composable
fun ServerStatusScreen() {
    Scaffold(
        containerColor = Color.White,
        topBar = {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = 48.dp, bottom = 16.dp, start = 24.dp, end = 24.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                IconButton(onClick = { }) {
                    Icon(Icons.Default.Menu, contentDescription = "Menu", tint = Color.Black)
                }
                Text(
                    text = "NodeLink",
                    fontSize = 22.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )
                Box {
                    IconButton(onClick = { }) {
                        Icon(Icons.Default.Notifications, contentDescription = "Alerts", tint = Color.Black)
                    }
                    Box(
                        modifier = Modifier
                            .size(10.dp)
                            .background(Color.Red, CircleShape)
                            .align(Alignment.TopEnd)
                            .padding(2.dp)
                    )
                }
            }
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(horizontal = 24.dp)
        ) {
// Hero Status Section
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(vertical = 16.dp),
                shape = RoundedCornerShape(24.dp),
                colors = CardDefaults.cardColors(containerColor = Color(0xFFF5F7FA))
            ) {
                Column(
                    modifier = Modifier.padding(24.dp),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Box(
                        modifier = Modifier
                            .size(100.dp)
                            .background(Color.White, CircleShape)
                            .padding(8.dp),
                        contentAlignment = Alignment.Center
                    ) {
                        Box(
                            modifier = Modifier
                                .fillMaxSize()
                                .background(Color(0xFFE8F5E9), CircleShape), // Light Green
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.Check,
                                contentDescription = "Active",
                                tint = Color(0xFF4CAF50),
                                modifier = Modifier.size(48.dp)
                            )
                        }
                    }
                    Spacer(modifier = Modifier.height(16.dp))
                    Text(
                        text = "System Operational",
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.Black
                    )
                    Text(
                        text = "Uptime: 14d 2h 15m",
                        fontSize = 14.sp,
                        color = Color.Gray,
                        modifier = Modifier.padding(top = 4.dp)
                    )

                    Spacer(modifier = Modifier.height(24.dp))

                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceEvenly
                    ) {
                        StatBadge(label = "CPU", value = "12%")
                        StatBadge(label = "RAM", value = "4.2GB")
                        StatBadge(label = "NET", value = "1Gbps")
                    }
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            // Quick Actions Grid
            Text(
                text = "Control Panel",
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black,
                modifier = Modifier.padding(bottom = 16.dp)
            )

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                ActionButton(
                    icon = Icons.Default.PlayArrow,
                    label = "Start",
                    color = Color(0xFFE3F2FD),
                    iconColor = Color(0xFF2196F3),
                    modifier = Modifier.weight(1f)
                )
                ActionButton(
                    icon = Icons.Default.Refresh,
                    label = "Reboot",
                    color = Color(0xFFFFF3E0),
                    iconColor = Color(0xFFFF9800),
                    modifier = Modifier.weight(1f)
                )
                ActionButton(
                    icon = Icons.Default.Lock,
                    label = "Lock",
                    color = Color(0xFFFFEBEE),
                    iconColor = Color(0xFFF44336),
                    modifier = Modifier.weight(1f)
                )
                ActionButton(
                    icon = Icons.Default.Settings,
                    label = "Config",
                    color = Color(0xFFF3E5F5),
                    iconColor = Color(0xFF9C27B0),
                    modifier = Modifier.weight(1f)
                )
            }

            Spacer(modifier = Modifier.height(32.dp))

            // Recent Logs
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Recent Logs",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )
                Text(
                    text = "View All",
                    fontSize = 14.sp,
                    color = Color(0xFF2196F3),
                    fontWeight = FontWeight.SemiBold,
                    modifier = Modifier.clickable { }
                )
            }

            Spacer(modifier = Modifier.height(8.dp))

            LazyColumn(
                modifier = Modifier.fillMaxWidth(),
                contentPadding = PaddingValues(bottom = 16.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                item {
                    LogItem(
                        icon = Icons.Default.Warning,
                        title = "High Memory Usage",
                        time = "10:42 AM",
                        typeColor = Color(0xFFFFF3E0), // Orange tint
                        iconTint = Color(0xFFFF9800)
                    )
                }
                item {
                    LogItem(
                        icon = Icons.Default.Person,
                        title = "Admin Login: root",
                        time = "09:15 AM",
                        typeColor = Color(0xFFE3F2FD), // Blue tint
                        iconTint = Color(0xFF2196F3)
                    )
                }
                item {
                    LogItem(
                        icon = Icons.Default.Info,
                        title = "Backup Completed",
                        time = "04:00 AM",
                        typeColor = Color(0xFFE8F5E9), // Green tint
                        iconTint = Color(0xFF4CAF50)
                    )
                }
            }
        }
    }
}
@Composable
fun StatBadge(label: String, value: String) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Text(
            text = value,
            fontWeight = FontWeight.Bold,
            fontSize = 16.sp,
            color = Color.Black
        )
        Text(
            text = label,
            fontSize = 12.sp,
            color = Color.Gray
        )
    }
}
@Composable
fun ActionButton(
    icon: ImageVector,
    label: String,
    color: Color,
    iconColor: Color,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(64.dp)
                .clip(RoundedCornerShape(16.dp))
                .background(color)
                .clickable { },
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = icon,
                contentDescription = label,
                tint = iconColor,
                modifier = Modifier.size(28.dp)
            )
        }
        Spacer(modifier = Modifier.height(8.dp))
        Text(
            text = label,
            fontSize = 12.sp,
            fontWeight = FontWeight.Medium,
            color = Color.Black
        )
    }
}
@Composable
fun LogItem(
    icon: ImageVector,
    title: String,
    time: String,
    typeColor: Color,
    iconTint: Color
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.White)
            .padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Box(
            modifier = Modifier
                .size(40.dp)
                .clip(CircleShape)
                .background(typeColor),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = iconTint,
                modifier = Modifier.size(20.dp)
            )
        }
        Spacer(modifier = Modifier.width(16.dp))
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = title,
                fontWeight = FontWeight.SemiBold,
                fontSize = 14.sp,
                color = Color.Black
            )
        }
        Text(
            text = time,
            fontSize = 12.sp,
            color = Color.Gray
        )
    }
}
@Preview(showBackground = true)
@Composable
fun PreviewServerStatus() {
    MaterialTheme {
        ServerStatusScreen()
    }
}

File: suc/swift/083.swift
import SwiftUI

@main
struct ServerStatusApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

struct ContentView: View {
    var body: some View {
        ServerStatusScreen()
            .preferredColorScheme(.light)
            .statusBarHidden(true)
    }
}

struct ServerStatusScreen: View {
    var body: some View {
        ZStack {
            Color.white.ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Top Bar
                TopBar()
                    .padding(.top, 48)
                    .padding(.horizontal, 24)
                    .padding(.bottom, 16)
                
                ScrollView(showsIndicators: false) {
                    VStack(alignment: .leading, spacing: 0) {
                        
                        // Hero Status Section
                        HeroStatusCard()
                            .padding(.vertical, 16)
                        
                        Spacer().frame(height: 16)
                        
                        // Quick Actions Grid
                        Text("Control Panel")
                            .font(.system(size: 18, weight: .bold))
                            .foregroundColor(.black)
                            .padding(.bottom, 16)
                        
                        HStack(spacing: 16) {
                            ActionButton(
                                iconName: "play.fill",
                                label: "Start",
                                color: Color(red: 0.89, green: 0.95, blue: 0.99), // #E3F2FD
                                iconColor: Color(red: 0.13, green: 0.59, blue: 0.95) // #2196F3
                            )
                            ActionButton(
                                iconName: "arrow.clockwise",
                                label: "Reboot",
                                color: Color(red: 1.00, green: 0.95, blue: 0.88), // #FFF3E0
                                iconColor: Color(red: 1.00, green: 0.60, blue: 0.00) // #FF9800
                            )
                            ActionButton(
                                iconName: "lock.fill",
                                label: "Lock",
                                color: Color(red: 1.00, green: 0.92, blue: 0.93), // #FFEBEE
                                iconColor: Color(red: 0.96, green: 0.26, blue: 0.21) // #F44336
                            )
                            ActionButton(
                                iconName: "gearshape.fill",
                                label: "Config",
                                color: Color(red: 0.95, green: 0.90, blue: 0.96), // #F3E5F5
                                iconColor: Color(red: 0.61, green: 0.15, blue: 0.69) // #9C27B0
                            )
                        }
                        
                        Spacer().frame(height: 32)
                        
                        // Recent Logs Header
                        HStack {
                            Text("Recent Logs")
                                .font(.system(size: 18, weight: .bold))
                                .foregroundColor(.black)
                            Spacer()
                            Button(action: {}) {
                                Text("View All")
                                    .font(.system(size: 14, weight: .semibold))
                                    .foregroundColor(Color(red: 0.13, green: 0.59, blue: 0.95)) // #2196F3
                            }
                        }
                        
                        Spacer().frame(height: 8)
                        
                        // Recent Logs List
                        VStack(spacing: 12) {
                            LogItem(
                                iconName: "exclamationmark.triangle.fill",
                                title: "High Memory Usage",
                                time: "10:42 AM",
                                typeColor: Color(red: 1.00, green: 0.95, blue: 0.88), // #FFF3E0
                                iconTint: Color(red: 1.00, green: 0.60, blue: 0.00) // #FF9800
                            )
                            LogItem(
                                iconName: "person.fill",
                                title: "Admin Login: root",
                                time: "09:15 AM",
                                typeColor: Color(red: 0.89, green: 0.95, blue: 0.99), // #E3F2FD
                                iconTint: Color(red: 0.13, green: 0.59, blue: 0.95) // #2196F3
                            )
                            LogItem(
                                iconName: "info.circle.fill",
                                title: "Backup Completed",
                                time: "04:00 AM",
                                typeColor: Color(red: 0.91, green: 0.96, blue: 0.91), // #E8F5E9
                                iconTint: Color(red: 0.30, green: 0.69, blue: 0.31) // #4CAF50
                            )
                        }
                        .padding(.bottom, 16)
                    }
                    .padding(.horizontal, 24)
                }
            }
        }
        .edgesIgnoringSafeArea(.all)
    }
}

struct TopBar: View {
    var body: some View {
        HStack {
            Button(action: {}) {
                Image(systemName: "line.3.horizontal")
                    .foregroundColor(.black)
                    .font(.system(size: 24))
            }
            Spacer()
            Text("NodeLink")
                .font(.system(size: 22, weight: .bold))
                .foregroundColor(.black)
            Spacer()
            ZStack(alignment: .topTrailing) {
                Button(action: {}) {
                    Image(systemName: "bell.fill")
                        .foregroundColor(.black)
                        .font(.system(size: 24))
                }
                Circle()
                    .fill(Color.red)
                    .frame(width: 10, height: 10)
                    .padding(2)
            }
        }
    }
}

struct HeroStatusCard: View {
    var body: some View {
        VStack {
            VStack {
                ZStack {
                    Circle()
                        .fill(Color.white)
                        .frame(width: 100, height: 100)
                    
                    ZStack {
                        Circle()
                            .fill(Color(red: 0.91, green: 0.96, blue: 0.91)) // #E8F5E9
                            .frame(width: 84, height: 84) // 100 - padding 8*2
                        
                        Image(systemName: "checkmark")
                            .font(.system(size: 40, weight: .bold)) // Approx 48dp visual
                            .foregroundColor(Color(red: 0.30, green: 0.69, blue: 0.31)) // #4CAF50
                    }
                }
                
                Spacer().frame(height: 16)
                
                Text("System Operational")
                    .font(.system(size: 20, weight: .bold))
                    .foregroundColor(.black)
                
                Text("Uptime: 14d 2h 15m")
                    .font(.system(size: 14))
                    .foregroundColor(.gray)
                    .padding(.top, 4)
                
                Spacer().frame(height: 24)
                
                HStack(spacing: 0) {
                    StatBadge(label: "CPU", value: "12%")
                        .frame(maxWidth: .infinity)
                    StatBadge(label: "RAM", value: "4.2GB")
                        .frame(maxWidth: .infinity)
                    StatBadge(label: "NET", value: "1Gbps")
                        .frame(maxWidth: .infinity)
                }
            }
            .padding(24)
        }
        .background(Color(red: 0.96, green: 0.97, blue: 0.98)) // #F5F7FA
        .cornerRadius(24)
    }
}

struct StatBadge: View {
    let label: String
    let value: String
    
    var body: some View {
        VStack(spacing: 2) {
            Text(value)
                .font(.system(size: 16, weight: .bold))
                .foregroundColor(.black)
            Text(label)
                .font(.system(size: 12))
                .foregroundColor(.gray)
        }
    }
}

struct ActionButton: View {
    let iconName: String
    let label: String
    let color: Color
    let iconColor: Color
    
    var body: some View {
        VStack {
            Button(action: {}) {
                ZStack {
                    RoundedRectangle(cornerRadius: 16)
                        .fill(color)
                        .frame(height: 64)
                    
                    Image(systemName: iconName)
                        .font(.system(size: 24)) // Approx 28dp visual
                        .foregroundColor(iconColor)
                }
            }
            Spacer().frame(height: 8)
            Text(label)
                .font(.system(size: 12, weight: .medium))
                .foregroundColor(.black)
        }
        .frame(maxWidth: .infinity)
    }
}

struct LogItem: View {
    let iconName: String
    let title: String
    let time: String
    let typeColor: Color
    let iconTint: Color
    
    var body: some View {
        HStack {
            ZStack {
                Circle()
                    .fill(typeColor)
                    .frame(width: 40, height: 40)
                
                Image(systemName: iconName)
                    .font(.system(size: 16)) // Approx 20dp visual
                    .foregroundColor(iconTint)
            }
            
            Spacer().frame(width: 16)
            
            Text(title)
                .font(.system(size: 14, weight: .semibold))
                .foregroundColor(.black)
            
            Spacer()
            
            Text(time)
                .font(.system(size: 12))
                .foregroundColor(.gray)
        }
        .padding(.vertical, 4)
        .background(Color.white)
    }
}